recipe-client-addon/lib/Sampling.jsm

```diff
@@ -57,25 +55,72 @@ function bufferToHex(buffer) {
   return hexCodes.join("");
 }
 
+function rangeBucketSample(inputHash, minBucket, maxBucket, bucketCount) {
+  const minHash = fractionToKey(minBucket / bucketCount);
+  const maxHash = fractionToKey(maxBucket / bucketCount);
+  return (minHash <= inputHash) && (inputHash < maxHash);
+}
+
+/**
+ * Return a promise for a truncated hash of `data`.
```

Comment:
The "Return a promise" bit reads weirdly to me, compared to other methods here that return promises and use `@promises` instead. I'm more interested that it hashes data, and what "truncated" means (as in truncated to 12 characters). I think you should reword this.